drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/src/main/java/org/drools/workbench/screens/scenariosimulation/backend/server/util/DMNSimulationCreationStrategy.java

```diff
@@ -161,12 +180,16 @@ private void addToScenario(FactMappingExtractor factMappingExtractor, FactModelT
                     previousSteps.add(factModelTree.getFactName());
                 }
                 previousSteps.add(entry.getKey());
-                addToScenario(factMappingExtractor, nestedModelTree, previousSteps, hiddenValues);
+
+                if (!alreadyVisited.contains(nestedModelTree.getFactName())) {
+                    alreadyVisited.add(factModelTree.getFactName());
+                    internalAddToScenario(factMappingExtractor, nestedModelTree, previousSteps, hiddenValues, alreadyVisited);
+                }
             }
         }
     }
 
-    static private class FactMappingExtractor {
+    static public class FactMappingExtractor {
```

Comment:
@danielezonca Please resolve this code smell: switch static and public keywords.

Reply:
Done